src/db/sync/state.rs

```diff
@@ -92,6 +59,75 @@ impl<Db: DatabaseRef + Sync> ConcurrentState<Db> {
         self.info.bundle_state.size_hint()
     }
 
+    /// State clear EIP-161 is enabled in Spurious Dragon hardfork.
+    pub fn set_state_clear_flag(&mut self, has_state_clear: bool) {
+        self.info.cache.set_state_clear_flag(has_state_clear);
+    }
+
+    /// Insert not existing account into cache state.
+    pub fn insert_not_existing(&mut self, address: Address) {
+        self.info.cache.insert_not_existing(address)
+    }
+
+    /// Insert account into cache state.
+    pub fn insert_account(&mut self, address: Address, info: AccountInfo) {
+        self.info.cache.insert_account(address, info)
+    }
+
+    /// Insert account with storage into cache state.
+    pub fn insert_account_with_storage(
+        &mut self,
+        address: Address,
+        info: AccountInfo,
+        storage: PlainStorage,
+    ) {
+        self.info.cache.insert_account_with_storage(address, info, storage)
+    }
+
+    /// Apply evm transitions to transition state.
+    pub fn apply_transition(&mut self, transitions: Vec<(Address, TransitionAccount)>) {
+        // add transition to transition state.
+        if let Some(s) = self.info.transition_state.as_mut() {
+            s.add_transitions(transitions)
+        }
+    }
+
+    /// Take all transitions and merge them inside bundle state.
+    /// This action will create final post state and all reverts so that
+    /// we at any time revert state of bundle to the state before transition
+    /// is applied.
+    pub fn merge_transitions(&mut self, retention: BundleRetention) {
+        if let Some(transition_state) = self.info.transition_state.take() {
+            self.info
+                .bundle_state
+                .apply_transitions_and_create_reverts(transition_state, retention);
+        }
+    }
+
+    // TODO make cache aware of transitions dropping by having global transition counter.
```

Comment:
let's ticket this?

Reply:
this is actually reproduced from revm. And I don't think they are planning on it anytime soon.